sklearn/cluster/k_means_.py

```diff
@@ -1556,7 +1556,7 @@ def fit(self, X, y=None, sample_weight=None):
                 init_size=init_size)
 
             # Compute the label assignment on the init dataset
-            batch_inertia, centers_squared_diff = _mini_batch_step(
+            _mini_batch_step(
```

Comment:
I mentioned this in the previous issue: is `n_iter` ever set to 0 for k-means, or is it strictly greater than 0? Because I would see how this assignment might be useful if the subsequent for loop does not assignment them.

Reply:
It looks contingent on `n_samples`, `self. max_iter` and `self. batch_size`

https://github.com/scikit-learn/scikit-learn/blob/2e2e69d412b205a6cb65dae964e6da2639e26803/sklearn/cluster/k_means_.py#L1524-L1525

So it appears very unlikely that `n_iter` would ever be set to 0.